Mod2/Lessons/Week2/CFUReview.md

```diff
@@ -0,0 +1,38 @@
+# Mod 2 Week 2 Checks for Understanding
+
+With your group, answer each question below.  Spend some time discussing each question _without_ using any resources; use your own recall of the topics covered this week.  If, after you have discussed without resources, your group needs assistance, use the raise hand feature and an instructor will come help!
+
+1. List at least three benefits of Test Driven Development (TDD).
+2. What could be some of the downsides of using TDD?
+3. True or False: In TDD you must write all of the tests for your feature first before doing any implementation. Explain your answer.
+4. You've teammate Omar is building using TDD and has asked you to give feedback on their test. Something important is missing, what's missing and why is it important?
+
+```c#
+public void IsCreatedWithCorrectAttributes()
+{
+    Customer customer = new Customer("Some Person", "123 A street, Bellingham, WA 98226");
+    Order order = new Order(customer);
+}
+
+```
+
+5. What is the difference between untit tests and integration tests?
+6. Which do you think it makes sense to write first, unit tests or integration tests?
+7. Here is the workflow Player 1 used during the Git for Pairs lab. How do you expect this workflow would change if you were working on a team of 4 instead of a team of 2? What extra difficulties might you run into when working with git in a team of 4 instead of a team of 2?
+
+Player 1 has the following responsibilies:
+  * Create a repository titled `FavoriteRestaurants`.
+  * Add a file that holds a list of everyone's favorite restaurant.
+  * Commit and push the main branch to Github.
+  * Add Player 2 as a collaborator.
+  * Create a new branch with your initials followed by the name (or abbreviation) of your favorite restaurant (e.g. `rt/red_lobster`).
+  * In your branch, add a separate file with information on your favorite restaurant: name, location, why it is your favorite restaurant, your favorite dish, etc.
+  * Add your name and the name of your favorite restaurant to the list.
+  * Commit and push your branch to Github.
+  * Make a PR for your branch.
+  * Review and merge the PR from Player 2. Fix any merge conflicts.
+  * After both Pull Requests have been merged, pull the `main` branch from Github.
+
+8. When working with Git in pairs, is it good practice to make commits on the main branch?
+9. Imagine that you're working with your partner Yavor on a project. You're both github collaborators on the project and you have each cloned down the repository from Github to your personal computers and are working in your local repositories. If Yavor creates a branch and adds some commits you won't be able to see his code. Why? What would he need to do for you to see what he's working on?
+10. Why is it good practice to have someone else approve and/or merge your PR?
```

Comment:
1 and 10 are very similar to questions from the assessment. Will probably just remove them and have 8 CFU quesitons.

Reply:
Sounds good to me!